plugin/pkg/scheduler/algorithm/listers.go

```diff
@@ -184,3 +185,42 @@ func (f FakeReplicaSetLister) GetPodReplicaSets(pod *v1.Pod) (rss []*extensions.
 
 	return
 }
+
+// StatefulSetLister interface represents anything that can produce a list of StatefulSet; the list is consumed by a scheduler.
+type StatefulSetLister interface {
+	// Gets the StatefulSet for the given pod.
+	GetPodStatefulSets(*v1.Pod) ([]*apps.StatefulSet, error)
+}
+
+// EmptyStatefulSetLister implements StatefulSetLister on []apps.StatefulSet returning empty data.
+type EmptyStatefulSetLister struct{}
+
+// GetPodStatefulSets of EmptyStatefulSetLister returns nil.
+func (f EmptyStatefulSetLister) GetPodStatefulSets(pod *v1.Pod) (sss []*apps.StatefulSet, err error) {
+	return nil, nil
+}
+
+// FakeStatefulSetLister implements ControllerLister on []apps.StatefulSet for testing purposes.
+type FakeStatefulSetLister []*apps.StatefulSet
```

Comment:
Typically Fake interfaces aren't in the mainline code where possible and are either implemented in the tests, or separated in some fashion.